state/relation.go

```diff
@@ -332,6 +332,41 @@ func (r *Relation) Destroy() error {
 	return err
 }
 
+// destroyCrossModelRelationUnitsOps returns the operations necessary for the units
+// of the specified remote application to leave scope.
+func destroyCrossModelRelationUnitsOps(op *ForcedOperation, remoteApp *RemoteApplication, rel *Relation, onlyForTerminated bool) ([]txn.Op, error) {
+	if onlyForTerminated {
+		statusInfo, err := remoteApp.Status()
+		if op.FatalError(err) && !errors.IsNotFound(err) {
+			return nil, errors.Trace(err)
+		}
+		if err != nil || statusInfo.Status != status.Terminated {
+			return nil, errors.Trace(err)
+		}
+	}
+	logger.Debugf("forcing cleanup of units for %v", remoteApp.Name())
+	remoteUnits, err := rel.AllRemoteUnits(remoteApp.Name())
+	if op.FatalError(err) {
+		return nil, errors.Trace(err)
+	}
```

Comment:
should at least handle the whole error (e.g. log, return) or add a comment.

Reply:
the error is collected in the `ForceOperation` error list for use later, but i can add extra logging here too